### PR TITLE
Merge: Double page pagination (#378)

### PR DESCRIPTION
**Pull request by matthias-ronge**
Adds a feature to generate double page counts on each image in the pagination. The available separators can be configured in `goobi_config.properties` and selected from a drop down list in the front end. Icons have been replaced to better represent what the function actually does.

unbenannt

### CODE REVIEW

File: Goobi/src/de/sub/goobi/metadaten/SelectOne.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
+package de.sub.goobi.metadaten;
+
```
Comment (henning-gerhardt) [inline Goobi/src/de/sub/goobi/metadaten/SelectOne.java]: Missing Goobi header

File: Goobi/src/de/sub/goobi/metadaten/Selectable.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package de.sub.goobi.metadaten;
+
```
Comment (henning-gerhardt) [inline Goobi/src/de/sub/goobi/metadaten/Selectable.java]: Missing Goobi header.

File: Goobi/src/de/sub/goobi/metadaten/Separator.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
+package de.sub.goobi.metadaten;
+
```
Comment (henning-gerhardt) [inline Goobi/src/de/sub/goobi/metadaten/Separator.java]: Missing Goobi header.

File: Goobi/newpages/images/buttons/paginierung_blatt.svg [hunk 4]
```diff
@@ -0,0 +1,140 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_blatt.svg]: Missing Goobi header.

File: Goobi/newpages/images/buttons/paginierung_blatt_inactive.svg [hunk 5]
```diff
@@ -0,0 +1,140 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_blatt_inactive.svg]: Missing Goobi header.

File: Goobi/newpages/images/buttons/paginierung_blatt_rectoverso.svg [hunk 6]
```diff
@@ -0,0 +1,142 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_blatt_rectoverso.svg]: Missing Goobi header.

File: Goobi/newpages/images/buttons/paginierung_blatt_rectoverso_inactive.svg [hunk 7]
```diff
@@ -0,0 +1,142 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_blatt_rectoverso_inactive.svg]: Missing Goobi header.

File: Goobi/newpages/images/buttons/paginierung_doppelseite.svg [hunk 8]
```diff
@@ -0,0 +1,167 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_doppelseite.svg]: Missing Goobi header.

File: Goobi/newpages/images/buttons/paginierung_doppelseite_inactive.svg [hunk 9]
```diff
@@ -0,0 +1,167 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_doppelseite_inactive.svg]: Missing Goobi header.

File: Goobi/newpages/images/buttons/paginierung_seite.svg [hunk 10]
```diff
@@ -0,0 +1,142 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_seite.svg]: Missing Goobi header.

File: Goobi/newpages/images/buttons/paginierung_seite_inactive.svg [hunk 11]
```diff
@@ -0,0 +1,142 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_seite_inactive.svg]: Missing Goobi header.

File: Goobi/newpages/images/buttons/paginierung_seite_rectoverso.svg [hunk 12]
```diff
@@ -0,0 +1,166 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_seite_rectoverso.svg]: Missing Goobi header.

File: Goobi/newpages/images/buttons/paginierung_seite_rectoverso_inactive.svg [hunk 13]
```diff
@@ -0,0 +1,166 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_seite_rectoverso_inactive.svg]: Missing Goobi header.

File: Goobi/newpages/images/buttons/paginierung_spalte.svg [hunk 14]
```diff
@@ -0,0 +1,162 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_spalte.svg]: Missing Goobi header.

File: Goobi/newpages/images/buttons/paginierung_spalte_inactive.svg [hunk 15]
```diff
@@ -0,0 +1,162 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_spalte_inactive.svg]: Missing Goobi header.

File: Goobi/newpages/images/buttons/paginierung_spalte_inactive.svg [hunk 16]
```diff
@@ -0,0 +1,162 @@
...
+         </g>
+        </g>
+       </g>
+      </g>
+     </g>
+    </g>
+   </g>
+  </g>
+ </g>
+</svg>
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_spalte_inactive.svg]: Please add new line at end of file.

File: Goobi/newpages/images/buttons/paginierung_blatt.svg [hunk 17]
```diff
@@ -0,0 +1,140 @@
...
+         </g>
+        </g>
+       </g>
+      </g>
+     </g>
+    </g>
+   </g>
+  </g>
+ </g>
+</svg>
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_blatt.svg]: Please add new line at end of file.

File: Goobi/newpages/images/buttons/paginierung_blatt_inactive.svg [hunk 18]
```diff
@@ -0,0 +1,140 @@
...
+         </g>
+        </g>
+       </g>
+      </g>
+     </g>
+    </g>
+   </g>
+  </g>
+ </g>
+</svg>
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_blatt_inactive.svg]: Please add new line at end of file.

File: Goobi/newpages/images/buttons/paginierung_blatt_rectoverso.svg [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+         </g>
+        </g>
+       </g>
+      </g>
+     </g>
+    </g>
+   </g>
+  </g>
+ </g>
+</svg>
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_blatt_rectoverso.svg]: Please add new line at end of file.

File: Goobi/newpages/images/buttons/paginierung_blatt_rectoverso_inactive.svg [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
+         </g>
+        </g>
+       </g>
+      </g>
+     </g>
+    </g>
+   </g>
+  </g>
+ </g>
+</svg>
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_blatt_rectoverso_inactive.svg]: Please add new line at end of file.

File: Goobi/newpages/images/buttons/paginierung_doppelseite.svg [hunk 21]
```diff
@@ -0,0 +1,167 @@
...
+         </g>
+        </g>
+       </g>
+      </g>
+     </g>
+    </g>
+   </g>
+  </g>
+ </g>
+</svg>
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_doppelseite.svg]: Please add new line at end of file.

File: Goobi/newpages/images/buttons/paginierung_doppelseite_inactive.svg [hunk 22]
```diff
@@ -0,0 +1,167 @@
...
+         </g>
+        </g>
+       </g>
+      </g>
+     </g>
+    </g>
+   </g>
+  </g>
+ </g>
+</svg>
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_doppelseite_inactive.svg]: Please add new line at end of file.

File: Goobi/newpages/images/buttons/paginierung_seite.svg [hunk 23]
```diff
@@ -0,0 +1,142 @@
...
+         </g>
+        </g>
+       </g>
+      </g>
+     </g>
+    </g>
+   </g>
+  </g>
+ </g>
+</svg>
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_seite.svg]: Please add new line at end of file.

File: Goobi/newpages/images/buttons/paginierung_seite_inactive.svg [hunk 24]
```diff
@@ -0,0 +1,142 @@
...
+         </g>
+        </g>
+       </g>
+      </g>
+     </g>
+    </g>
+   </g>
+  </g>
+ </g>
+</svg>
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_seite_inactive.svg]: Please add new line at end of file.

File: Goobi/newpages/images/buttons/paginierung_seite_rectoverso.svg [hunk 25]
```diff
@@ -0,0 +1,166 @@
...
+         </g>
+        </g>
+       </g>
+      </g>
+     </g>
+    </g>
+   </g>
+  </g>
+ </g>
+</svg>
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_seite_rectoverso.svg]: Please add new line at end of file.

File: Goobi/newpages/images/buttons/paginierung_seite_rectoverso_inactive.svg [hunk 26]
```diff
@@ -0,0 +1,166 @@
...
+         </g>
+        </g>
+       </g>
+      </g>
+     </g>
+    </g>
+   </g>
+  </g>
+ </g>
+</svg>
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_seite_rectoverso_inactive.svg]: Please add new line at end of file.

File: Goobi/newpages/images/buttons/paginierung_spalte.svg [hunk 27]
```diff
@@ -0,0 +1,162 @@
...
+         </g>
+        </g>
+       </g>
+      </g>
+     </g>
+    </g>
+   </g>
+  </g>
+ </g>
+</svg>
```
Comment (henning-gerhardt) [inline Goobi/newpages/images/buttons/paginierung_spalte.svg]: Please add new line at end of file.

**Comment by stweil** [comment]
@henning-gerhardt, depending on your SVG editor, it might be difficult to add the Goobi header. Of course you can always use a text editor, but that's not the normal way how SVG graphics are handled...

I know that the coding guidelines require that header, but I think we should interpret the rules and apply them where it makes sense.

**Comment by henning-gerhardt** [comment]
Your changes are breaking test `de.sub.goobi.metadaten.PaginatorTest`! 

Please don't remove one of the few existing tests as a solution for this broken test.

**Comment by henning-gerhardt** [comment]
One question: is page separator always visible or only visible for double page pagination?

**Comment by matthias-ronge** [comment]
Would it be okay to treat the SVG files as binaries, even though they are showing as text on GitHub? They have been created as-is from LibreOffice Draw image export. I don’t write images manually. I tried putting the license header into the file properties description comment field, but that does not get written out in SVG export. Also opening and saving the file again results in completely different results in a way that is virtually impossible to comprehend if you try to diff them. Also this would increase each page load by about 10 kB (6 images). Is this necessary?

Secondly, the GPL was built to target program code specifically. We should not use it to license *images*, should we? Some CC license would be more appropriate.

Third, *if* we add the default license header to the SVG XML, does it require to contain the “linking excption”? An image is not program code.

**Comment by matthias-ronge** [comment]
Page separator drop-down select shall only be visible in the two double page pagination modes and be hidden in the four single page modes.

**Comment by henning-gerhardt** [comment]
- regarding svg files: You convinced me to treat this files as binary files, so no license text is needed

- one more thing: default value for `pageSeparators` should only be a white space character to match current behaviour. If somebody use double page pagination he / she must adjust this value explicit to theirs requirements. Aside the fact to produce non valid DFG conform METS/MODS files on export.

**Comment by matthias-ronge** [comment]
The default of `pageSeparators` *is* defined as a single white space in [Metadaten.java, ll. 2014 f.](https://github.com/matthias-ronge/goobi-production/blob/issue_378/Goobi/src/de/sub/goobi/metadaten/Metadaten.java#L2014-L2015) The [usage example](https://github.com/matthias-ronge/goobi-production/blob/issue_378/Goobi/config/goobi_config.properties#L274-L277) in `goobi_config.properties` is there to demonstrate the syntax and its possibilities but there is no need to use it. Just comment it out to get the default behaviour working.

White space is showing as an *open box* (“␣”, U+2423) in the drop down select to make the different variants more clear, think of `;` /  ` ; ` / `; ` vs. `;` /  `␣;␣` / `;␣`.

**Comment by henning-gerhardt** [comment]
In my opinion is a defined default value in source code only a fall back behaviour for a working program. I will change this configuration parameter in my installation.

**Comment by matthias-ronge** [comment]
What would you propose?

**Comment by henning-gerhardt** [comment]
As mentioned above I would change value of `pageSeparators` to `" "`: (`pageSeparators=" "`) on my installation. Your complex value could be used as example:

```
# Comma-separated list of separators available for pagination modes where two
# pages are on one image. Enclose in double quotes if contains white space or
# comma. For example
# pageSeparators=" ",",",;," , "," ; "," / "
# Defaults to one single white space:
pageSeparators=" "
```

**Comment by henning-gerhardt** [comment]
Looks now good for me. I will merge this later today or tomorrow.

**Pull request merged**